### PR TITLE
mzcompose: typecheck all mzworkflows.py files

### LINKED ISSUE #5022: mzconduct generating "WARNING: unknown env var" messages
When running mzcompose, I get the following error messages:

```
C> WARNING: unknown env var 'DEFAULT_PROGRESS_MODE'
C> WARNING: unknown env var 'API_KEY'
C> WARNING: unknown env var 'ARCHIVE_PATH'
```

Looking at `ARCHIVE_PATH` for example, it's only present in the `play/mbta` demo:

```
play/mbta/doc/mbta-setup.md
88:`ARCHIVE_PATH=<insert_path_archive> ../../bin/mzconduct run mbta -w replay`. Do not uncompress
the archive.

play/mbta/mzcompose.yml
72:          ${ARCHIVE_PATH}
```

However, this message prints even when running `mzconduct` for other workflows:

```
[chris-work:~/materialize/materialize:(wikirecent-demo[|])] ./bin/mzconduct web wikirecent app
C> WARNING: unknown env var 'DEFAULT_PROGRESS_MODE'
C> WARNING: unknown env var 'API_KEY'
C> WARNING: unknown env var 'ARCHIVE_PATH'
```

Should we make this an error message? Should I be getting this warning message even when running other
workflows or should we only print this message if it's the current workflow?

@quodlibetor, thoughts?

### PR DESCRIPTION
**Pull request by benesch**
Several commits in here that remove support for the old YAML-based mzcompose workflows in order to get typechecking on all `mzworkflows.py` files in the repository. 

Touches #9809.
Incidentally fixes #5022 by removing the environment variable interpolation features.

### Motivation

   * This PR refactors existing code.

### Tips for reviewer

I was very careful to split this up into commits to make it reviewable. I strongly suggest you go commit by commit or it will look much scarier than it is! Commits that are just code movement are noted as such in their commit messages.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/9918)

### CODE REVIEW

File: bin/pycheck [hunk 1]
```diff
@@ -20,6 +20,14 @@ cd "$(dirname "$0")/.."
 py_folders=(ci misc/python)
 
 try bin/pyactivate --dev -m mypy "${py_folders[@]}"
+if ! try_last_failed; then
+    # mzworkflows.py files need to be passed to mypy individually to avoid
+    # duplicate module warnings. We don't check these if the previous `mypy`
+    # invocation failed, though, or we'll get a ton of duplicate warnings.
+    for file in $(git_files "**/mzworkflows.py"); do
```
Comment (philip-stoev) [inline bin/pycheck:27]: Can we just typecheck all the python stuff in test ? I can do this in  a follow-up PR .

File: misc/python/materialize/mzcompose/__init__.py [hunk 2]
```diff
@@ -577,281 +576,142 @@ def start_and_wait_for_tcp(self, services: List[str]) -> None:
         # TODO(benesch): once the workflow API is a proper Python API,
```
Comment (philip-stoev) [inline misc/python/materialize/mzcompose/__init__.py:576]: This comment is no longer relevant, should be removed.

**Review by philip-stoev — APPROVED** [review]
Thanks a lot for this!

Nightly is happy https://buildkite.com/materialize/nightlies/builds/564

so I think everything is in order. I reviewed this per-commit, with the entire diff and then read through `mzcompose/__init__.py` in the editor.

**Comment by benesch** [comment]
🎉 thanks very much for looking, @philip-stoev!

**Pull request merged**